staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -235,6 +243,8 @@ func (pf *PortForwarder) forward() error {
 	// wait for interrupt or conn closure
 	select {
 	case <-pf.stopChan:
+	case <-pf.streamFailChan:
+		return ErrLostConnectionToPod
```

Comment:
drop `streamFailChan` and drop this case, `<-pf.streamConn.CloseChan()` should be sufficient

Reply:
I've tried w/o that channel, and I was hoping that `<-pf.streamConn.CloseChan()` will be sufficient after we close the `_ = pf.streamConn.Close()` but after I moved it from error handling to after stream creation it does not. That's why I introduced another channel.

Reply:
> after I moved it from error handling to after stream creation it does not

uh... that makes me lose confidence in this relocation as a backstop... we have to understand why that is no longer sufficient to dump out of the loop and tear down the overall connection